### ISSUE TITLE
[rom_ext] Add device_id to unlock owner operation challenge request

### ISSUE THREAD

**Issue by jettr** [lowRISC/opentitan#24552]
### Description

It would be nice to ensure that the unlock request for a device is locked to a specific device id in addition to protecting it against replay attacks (e.g. nonce).

Having the device id part for the unlock challenge that needs to be signed by the owner keys also helps with record keeping and ensuring that a request is valid for only a specific device. This also allows the ability for infrastructure to collect votes based on device id, and then the infrastructure will sign any nonce value.

**Comment by jettr** [comment]
This was completely right?

**Comment by cfrantz** [comment]
Yes, this was completed by 0a8689fb2b32fa0b4519479a785fb691ff179933
